tests/components/bayesian/test_binary_sensor.py

```diff
@@ -131,9 +190,62 @@ def test_sensor_state(self):
         self.hass.block_till_done()
 
         state = self.hass.states.get("binary_sensor.test_binary")
-        assert [{"prob_true": 0.8, "prob_false": 0.4}] == state.attributes.get(
-            "observations"
-        )
+        assert state.attributes.get("observations")[0]["prob_given_true"] == 0.8
+        assert state.attributes.get("observations")[0]["prob_given_false"] == 0.4
+        assert round(abs(0.33 - state.attributes.get("probability")), 7) == 0
+
+        assert state.state == "on"
+
+        self.hass.states.set("sensor.test_monitored", "off")
+        self.hass.block_till_done()
+        self.hass.states.set("sensor.test_monitored", "on")
+        self.hass.block_till_done()
+
+        state = self.hass.states.get("binary_sensor.test_binary")
+        assert round(abs(0.2 - state.attributes.get("probability")), 7) == 0
+
+        assert state.state == "off"
+
+    def test_sensor_value_template(self):
+        """Test sensor on template platform observations."""
+        config = {
+            "binary_sensor": {
+                "name": "Test_Binary",
+                "platform": "bayesian",
+                "observations": [
+                    {
+                        "platform": "template",
+                        "value_template": "{{states('sensor.test_monitored') == 'off'}}",
+                        "prob_given_true": 0.8,
+                        "prob_given_false": 0.4,
+                    }
+                ],
+                "prior": 0.2,
+                "probability_threshold": 0.32,
+            }
+        }
+
+        assert setup_component(self.hass, "binary_sensor", config)
+
+        self.hass.states.set("sensor.test_monitored", "on")
+
+        state = self.hass.states.get("binary_sensor.test_binary")
+
+        assert [None] == state.attributes.get("observations")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5279916)